app/components/formatted_time.js

```diff
@@ -3,35 +3,36 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import 'date-time-format-timezone/build/src/date-time-format-timezone-all-zones-no-locale';
+import {injectIntl, intlShape} from 'react-intl';
 import {Text} from 'react-native';
 
 export default class FormattedTime extends React.PureComponent {
     static propTypes = {
         value: PropTypes.any.isRequired,
+        format: PropTypes.string,
+        timeZone: PropTypes.string,
         children: PropTypes.func,
         hour12: PropTypes.bool,
     };
 
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
     render() {
         const {
             value,
             children,
-            hour12,
+            timeZone,
+            ...props
         } = this.props;
+        const {intl} = this.context;
 
-        const date = new Date(value);
-        const militaryTime = !hour12;
-
-        const hour = militaryTime ? date.getHours() : (date.getHours() % 12 || 12);
-        let minute = date.getMinutes();
-        minute = minute >= 10 ? minute : ('0' + minute);
-        let time = '';
-
-        if (!militaryTime) {
-            time = (date.getHours() >= 12 ? ' PM' : ' AM');
-        }
+        Reflect.deleteProperty(props, 'format');
 
-        const formattedTime = hour + ':' + minute + time;
+        const timezoneProps = timeZone ? {timeZone} : {};
+        const formattedTime = intl.formatDate(value, {...props, ...timezoneProps, hour: 'numeric', minute: 'numeric'});
```

Comment:
I was going to propose switching to `intl.formatTime` here, but it doesn't look like that's necessary since the methods are almost identical

https://github.com/yahoo/react-intl/blob/master/src/format.js#L77